mllib/src/main/scala/org/apache/spark/ml/tree/treeParams.scala

```diff
@@ -360,7 +362,8 @@ private[ml] trait RandomForestParams extends TreeEnsembleParams {
     "The number of features to consider for splits at each tree node." +
       s" Supported options: ${RandomForestParams.supportedFeatureSubsetStrategies.mkString(", ")}",
     (value: String) =>
-      RandomForestParams.supportedFeatureSubsetStrategies.contains(value.toLowerCase))
+      RandomForestParams.supportedFeatureSubsetStrategies.contains(value.toLowerCase)
+      || value.matches("^(?:[1-9]\\d*|0\\.\\d*[1-9]\\d*\\d*|1\\.0+)$"))
```

Comment:
Yes, that sounds best to me. Also, I am not an expert in regex, but in this pattern `0\\.\\d*[1-9]\\d*\\d*` is the last `\\d*` redundant? I also think that you should be allowed to set the fraction with `".25"` but that doesn't work currently. Can we change the middle option to be `"0?\\.\\d*[1-9]\\d*"`